### PR TITLE
Add missing association to support Tiered Pricing

### PR DESCRIPTION
**Pull request by joannasese**
Establish `has_many :tiers` association in AddOn class

Include `changed_attributes()` to enable update functionality

### CODE REVIEW

File: lib/recurly/add_on.rb [hunk 1]
```diff
@@ -21,7 +22,7 @@ class AddOn < Resource
       created_at
       updated_at
       tier_type
-      tiers
+      # tiers
```
Comment (bhelx) [inline lib/recurly/add_on.rb]: This results in a behavior I don't think you expected:

```
irb(main):001:0> x = %w(
irb(main):002:0]   one
irb(main):003:0]   two
irb(main):004:0]   # three)
=> ["one", "two", "#", "three"]
```

Comment (joannasese) [reply lib/recurly/add_on.rb]: nope absolutely not!

File: lib/recurly/tier.rb [hunk 2]
```diff
@@ -8,6 +8,10 @@ class Tier < Resource
       unit_amount_in_cents
     )
 
+    def xml_keys
```
Comment (bhelx) [inline lib/recurly/tier.rb:11]: Why is this needed?

Comment (joannasese) [reply lib/recurly/tier.rb:11]: If I have an existing add-on and I want to update it, I could send in a request like this:
```ruby
add_on.tiers[1].unit_amount_in_cents = 2222
add_on.save!
```

Request body with `xml_keys()`:
```ruby
<add_on>
  <tiers>
    <tier>
      <ending_quantity>25</ending_quantity>
      <unit_amount_in_cents><USD>4560</USD></unit_amount_in_cents>
    </tier>
    <tier>
      <ending_quantity>999999999</ending_quantity>
      <unit_amount_in_cents><USD>2222</USD></unit_amount_in_cents>
    </tier>
  </tiers>
</add_on>
```

Without it:
```ruby
<add_on>
  <tiers>
    <tier/>
    <tier>
      <unit_amount_in_cents><USD>2222</USD></unit_amount_in_cents>
    </tier>
  </tiers>
</add_on>
```

`xml_keys()` ensures that `tier` is populated with required values. Otherwise, the errors below are triggered:
```ruby
<errors>
  <error field="add_on.tiers[ending_quantity=999999999].currencies" symbol="blank">can't be blank</error>
  <error field="add_on.tiers" symbol="not_unique">must have unique ending_quantities</error>
</errors>
```
